sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/SynchronousReceiver.java

```diff
@@ -164,7 +164,10 @@ private Flux<ServiceBusReceivedMessage> traceDecorator(Flux<ServiceBusReceivedMe
         final Flux<ServiceBusReceivedMessage> decorated = tracer.traceSyncReceive(SYNC_RECEIVE_SPAN_NAME, toDecorate);
         // TODO (anu) - discuss with Liudmila - do we need decorated.subscribe() here or IterableStream's internal
         //              subscription to the 'decorated' flux will do?
-        decorated.subscribe();
+        decorated.subscribe(i -> {
```

Comment:
Not for this PR, but is there some general approach we can do on when we should do it? We have similar code in what feels like random places, could we do it in some wholistic way?

also maybe we can have a helper method just for the sake of explaining what this thing does, like 

```java
  Flux<T> dropError(Flux<T> f) {
      f.subscribe(...);
  | 
```

Reply:
I think it comes down to two main cases-
1. If we are certain that the error will be reported or inspected downstream through a different asynchronous route, it is safe to add no-op handlers to these subscribers.
2. If there is a possibility that the error might be lost in the asynchronous chain, for instance, due to a timeout or closure of the parent channel, then it is useful to incorporate handlers with WARN logs for troubleshooting purposes.

We should analyze the code again and could use util as you suggested to centralize it, we could do something like –
```java
  Subscription subscribe(Publisher<T> p, ClientLogger logger) {
      If (logger == null) {
         return p.subscribe(no-op-handlers)
      } else {
        return p.subscribe(warn-log-handlers)
      }
  }
```
I’ll open an issue

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/42807